src/pages/iou/IOUDetailsModal.js

```diff
@@ -0,0 +1,158 @@
+import compose from '../../libs/compose';
+
+const propTypes = {
+    // URL Route params
+    route: PropTypes.shape({
+        // Params from the URL path
+        params: PropTypes.shape({
+            // chatReportID passed via route: /iou/details/:chatReportID/:iouReportID
+            chatReportID: PropTypes.string,
+
+            // iouReportID passed via route: /iou/details/:chatReportID/:iouReportID
+            iouReportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /* Onyx Props */
+    // Holds data related to IOU view state, rather than the underlying IOU data.
+    iou: PropTypes.shape({
+        // Is the IOU Report currently being paid?
+        loading: PropTypes.bool,
+
+        // Error message, empty represents no error
+        error: PropTypes.string,
+    }),
+
+    // IOU Report data object
+    iouReport: PropTypes.shape({
+        // ID for the chatReport that this IOU is linked to
+        chatReportID: PropTypes.number,
+
+        // Manager is the person who currently owes money
+        managerEmail: PropTypes.string,
+
+        // Owner is the person who is owed money
+        ownerEmail: PropTypes.string,
+
+        // The IOU transactions
+        transactions: PropTypes.arrayOf(PropTypes.shape(iouTansactionPropTypes)),
+
+        // Does the report have an outstanding IOU that needs to be paid?
+        hasOutstandingIOU: PropTypes.bool,
+    }).isRequired,
+
+    // Session info for the currently logged in user.
+    session: PropTypes.shape({
+        // Currently logged in user email
+        email: PropTypes.string,
+    }).isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    iou: {},
+};
+
+class IOUDetailsModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            // Temporarily placeholder, as the Venmo/Paypal issue will introduce settlement types.
+            settlementType: 'Elsewhere',
+        };
+
+        this.performIOUSettlement = this.performIOUSettlement.bind(this);
+    }
+
+    componentDidMount() {
+        // Fetch the iouReport without linking it to the chatReport. Else, the chatReport's iouReportID field would
+        // wrongly be pointed to an old paid iouReport, overwriting the iouReportID of the open iouReport.
+        fetchIOUReportByID(this.props.route.params.iouReportID, this.props.route.params.chatReportID);
+    }
+
+    performIOUSettlement() {
+        payIOUReport({
+            chatReportID: this.props.route.params.chatReportID,
+            reportID: this.props.route.params.iouReportID,
+            paymentMethodType: this.state.settlementType,
+        });
+    }
+
+    render() {
+        const sessionEmail = lodashGet(this.props.session, 'email', null);
+        const reportIsLoading = _.isUndefined(this.props.iouReport);
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title="Details"
```

Comment:
Please use the translate method of the withLocalize HOC

Reply:
Done, please re-review to confirm.